ads/model/model_description.py

```
@@ -0,0 +1,173 @@
+import json
+import ads.common
+import oci
+import pytz
+import datetime
+import os
+from oci.data_science.models import Metadata
+import ads
+
+class ModelDescription:
+
+    region = ''
+
+    empty_json = {
+        "version": "1.0",
+        "type": "modelOSSReferenceDescription",
+        "models": []
+    }
+    
+    def auth(self):        
+        authData = ads.common.auth.default_signer()
+        signer = authData['signer']
+        self.region = authData['config']['region']
+
+        # data science client
+        self.data_science_client = oci.data_science.DataScienceClient({'region': self.region}, signer=signer)
+        # oss client
+        self.object_storage_client = oci.object_storage.ObjectStorageClient({'region': self.region}, signer = signer)
+    
+    def __init__(self, model_ocid=None):
+
+        self.auth()
+
+        if model_ocid == None: 
+            # if no model given then start from scratch
+            self.modelDescriptionJson = self.empty_json
+        else:
+            # if model given then get that as the starting reference point
+            print("Getting model details from backend")
+            destination_file_path = "downloaded_artifact.json"
+            get_model_artifact_content_response = self.data_science_client.get_model_artifact_content(
+                model_id=model_ocid,
+            )
+            try:
+                with open(destination_file_path, "wb") as f:
+                    f.write(get_model_artifact_content_response.data.content)
+                with open(destination_file_path, 'r') as f:
+                    self.modelDescriptionJson = json.load(f)
+            except FileNotFoundError:
```

we should just let the exception propagate up - this is too much like Java :)

---

Yes, thanks for the comment. Now have raised errors everywhere needed.

Note for this code piece: Now I have changed the code a bit, instead of populating the file with API response and then reading the file contents into a local variable, I have skipped the file part.